src/client/Client.cpp

```diff
@@ -3621,3 +3621,17 @@ bool Client::verifyVolumeID(const std::string& id){
 		return false;
 	return true;
 }
+
+bool Client::verifyNamespaceName(const std::string& namespaceName){
+	if (namespaceName.length() <= LABEL_NAME_MAX_LEN){
+		if (std::iswalnum(namespaceName.front()) && std::iswalnum(namespaceName.back())) {
+			for (auto itr = std::next(namespaceName.begin()); itr != std::prev(namespaceName.end()); ++itr) {
```

Comment:
I think you'll want to be a bit careful with the logic here; if `namespaceName` is only a single character the two iterators (`next(begin)` and `prev(end)`) can pass each other without triggering the loop's termination condition and Bad Things will happen. I think the easiest way to fix this is to add a check just before the loop which immedaitely returns `true` if `namespaceName.size()` is exactly `1`.

Reply:
Fixed in latest commit.